sync/peerset/peer/metric/metric.go

```diff
@@ -0,0 +1,54 @@
+package metric
+
+import "github.com/pactus-project/pactus/sync/bundle/message"
+
+type Counter struct {
+	Bytes   int64
+	Bundles int64
+}
+
+type Metric struct {
+	TotalInvalid    Counter
+	TotalSent       Counter
+	TotalReceived   Counter
+	MessageSent     map[message.Type]*Counter
+	MessageReceived map[message.Type]*Counter
+}
+
+func NewMetric() Metric {
+	return Metric{
```

Comment:
Used pointer receiver for methods, better return pointer.


```suggestion
	return *Metric{
```

Reply:
To avoid memory fragmentation in the peerset, it's better not to use pointers.